packages/fast-components-styles-msft/src/select/index.ts

```diff
@@ -13,59 +17,78 @@ import {
 } from "../utilities/color";
 import { applyFloatingCornerRadius } from "../utilities/border";
 import { designUnit } from "../utilities/design-system";
+import { inputFieldStyles } from "../patterns/input-field";
 
-export const selectDisplayButtonOverrides: ComponentStyles<
-    Partial<ButtonClassNameContract>,
-    DesignSystem
-> = {
-    button: {
-        width: "100%",
-        padding: "0 10px",
-    },
-    button_contentRegion: {
-        width: "100%",
-        display: "inline-flex",
-    },
-};
+const styles: ComponentStyles<SelectClassNameContract, DesignSystem> = (
+    config: DesignSystem
+): ComponentStyleSheet<SelectClassNameContract, DesignSystem> => {
+    const designSystem: DesignSystem = withDesignSystemDefaults(config);
+    const direction: Direction = designSystem.direction;
+
+    return {
+        select: {
+            minWidth: "276px",
+            maxWidth: "374px",
+        },
+
+        select_button: {
+            height: height(),
+            width: "100%",
+            ...inputFieldStyles(),
+        },
 
-const styles: ComponentStyles<SelectClassNameContract, DesignSystem> = {
-    select: {
-        minWidth: "276px",
-        maxWidth: "374px",
-    },
+        select_buttonContentRegion: {
+            display: "grid",
+            gridTemplateColumns: "1fr auto",
+            alignItems: "center",
+            justifyItems: "start",
+        },
 
-    select_toggleGlyph: {
-        fill: neutralForegroundRest,
-    },
+        select_buttonDisplayText: {
+            ...ellipsis(),
+            textAlign: applyLocalizedProperty("left", "right", direction),
+            width: "100%",
+        },
 
-    select__disabled: {
-        opacity: ".3",
-    },
+        select_toggleGlyph: {
+            [applyLocalizedProperty(
+                "margin-left",
+                "margin-right",
+                direction
+            )]: horizontalSpacing(),
+            fill: neutralForegroundRest,
+            width: glyphSize,
+            height: glyphSize,
+            gridColumnStart: "2",
+        },
 
-    select_menu: {
-        background: neutralFillStealthRest,
-        ...applyFloatingCornerRadius(),
-        ...applyElevation(ElevationMultiplier.e11),
-        zIndex: "1",
-        position: "absolute",
-        width: "100%",
-        margin: "0",
-        padding: format("{0} 0", toPx<DesignSystem>(designUnit)),
-        maxWidth: "374px",
-        minWidth: "276px",
-        maxHeight: "328px",
-        overflow: "auto",
-    },
+        select__disabled: {},
```

Comment:
are there no disabled styles?

Reply:
We get those from ...inputFieldStyles() under "select_button".  So this implementation doesn't require this in the stylesheet so removed.